skyportal/handlers/api/telescope.py

```diff
@@ -135,23 +135,32 @@ def get(self, telescope_id=None):
         telescopes = []
         for telescope in data:
             temp = telescope.to_dict()
-            if telescope.next_twilight_morning_astronomical() is not None:
-                temp['is_night_astronomical'] = bool(
-                    telescope.next_twilight_morning_astronomical().jd
-                    < telescope.next_twilight_evening_astronomical().jd
-                )
-                temp[
-                    'next_twilight_morning_astronomical'
-                ] = telescope.next_twilight_morning_astronomical().iso
-                temp[
-                    'next_twilight_evening_astronomical'
-                ] = telescope.next_twilight_evening_astronomical().iso
-            else:
-                temp['is_night_astronomical'] = False
-                temp['next_twilight_morning_astronomical'] = False
-                temp['next_twilight_evening_astronomical'] = False
-
+            morning = False
+            evening = False
+            is_night_astronomical = False
+            if (
+                telescope.fixed_location
+                and telescope.lon
+                and telescope.lat
+                and telescope.elevation
+            ):
+                try:
+                    morning = telescope.next_twilight_morning_astronomical()
+                    evening = telescope.next_twilight_evening_astronomical()
+                    if morning is not None and evening is not None:
+                        is_night_astronomical = bool(morning.jd < evening.jd)
+                        morning = morning.iso
+                        evening = evening.iso
+                except Exception:
+                    morning = False
+                    evening = False
+                    is_night_astronomical = False
+                    continue
```

Comment:
Did you mean to break the loop with a continue?

Reply:
No ! I meant to just continue without raising an error, should i use `pass` instead ?

Reply:
ah yeah pass is what you want then, but you shouldn't need anything.

Reply:
pass is a no-op, so only used in the context of defining an empty function

Reply:
Michael is correct that the continue should be removed

Reply:
By catching the exception, you are already suppressing it. You can re-raise an exception from a catch clause, if that is needed, otherwise operations will continue without any. Another useful trick is the finally clause, which will execute regardless of whether an exception was raised or not.

Reply:
Ok ! I just removed the `continue` then.